libraries/DataFlash/LogFile.cpp

```diff
@@ -1926,3 +1926,80 @@ void DataFlash_Class::Log_Write_Beacon(AP_Beacon &beacon)
+        engine_speed_rpm                        : first_efi_state->engine_speed_rpm,  
+        spark_dwell_time_ms                     : first_efi_state->spark_dwell_time_ms,
+        atmospheric_pressure_kpa                : first_efi_state->atmospheric_pressure_kpa,
+        intake_manifold_pressure_kpa            : first_efi_state->intake_manifold_pressure_kpa,
+        intake_manifold_temperature             : first_efi_state->intake_manifold_temperature,  
+        coolant_temperature                     : first_efi_state->coolant_temperature,
+        oil_pressure                            : first_efi_state->oil_pressure,             
+        oil_temperature                         : first_efi_state->oil_temperature,
+        fuel_pressure                           : first_efi_state->fuel_pressure,
+        fuel_consumption_rate_cm3pm             : first_efi_state->fuel_consumption_rate_cm3pm,                    
+        estimated_consumed_fuel_volume_cm3      : first_efi_state->estimated_consumed_fuel_volume_cm3, 
+        throttle_position_percent               : first_efi_state->throttle_position_percent,            
+        ecu_index                               : first_efi_state->ecu_index
+    };
+
+    struct log_EFI2 pkt_efi2 = {
+        LOG_PACKET_HEADER_INIT(LOG_EFI2_MSG),
+        time_us                     : AP_HAL::micros64(),
+        health                      : AP_EFI::is_healthy(*first_efi_state),
+        engine_state                : first_efi_state->engine_state,
+        general_error               : first_efi_state->general_error,
+        crankshaft_sensor_status    : first_efi_state->crankshaft_sensor_status,
+        temperature_status          : first_efi_state->temperature_status,
+        fuel_pressure_status        : first_efi_state->fuel_pressure_status,
+        oil_pressure_status         : first_efi_state->oil_pressure_status,
+        detonation_status           : first_efi_state->detonation_status,
+        misfire_status              : first_efi_state->misfire_status,
+        debris_status               : first_efi_state->debris_status,
+        spark_plug_usage            : first_efi_state->spark_plug_usage,
+        ecu_index                   : first_efi_state->ecu_index
+    };
+
+    WriteBlock(&pkt_efi, sizeof(pkt_efi));
+    WriteBlock(&pkt_efi2, sizeof(pkt_efi2));
+
+    // Write data for a max of 4 cylinders
+    for (int i = 0; i < 4; i++) {
+        LogMessages msg_type;
+        switch (i) {
+            case 0:
+                msg_type = LOG_EFI_CYL1_MSG;
+                break;
+            case 1:
+                msg_type = LOG_EFI_CYL2_MSG;
+                break;
+            case 2:
+                msg_type = LOG_EFI_CYL3_MSG;
+                break;
+            case 3:
+                msg_type = LOG_EFI_CYL4_MSG;
+                break;
+        }
+        
+        struct log_EFI_CYL pkt_efi_cyl = {
+            LOG_PACKET_HEADER_INIT(msg_type),
```

Comment:
You can skip this whole switch thing by using this one line:
`LOG_PACKET_HEADER_INIT((uint8_t)(LOG_EFI_CYL1_MSG + i))`